app/objects/c_source.py

```diff
@@ -44,6 +44,17 @@ def fix_adjustments(self, in_data, **_):
                         x.append(dict(ability_id=ability_id, trait=trait, value=change.get('value'),
                                       offset=change.get('offset')))
         in_data['adjustments'] = x
+        if 'facts' in in_data:
```

Comment:
`fix_adjustments()` is called when the source is ingested at server boot (this is marshmallow's `pre_load`, which runs when you load something). This code here is just used to make sure that CALDERA properly initializes facts/relationship objects in the source file being created, rather than just leaving them as raw dictionaries (at least, I think they're left as raw dictionaries otherwise - they might be raw strings).